recipes/doxygen/all/conanfile.py

```diff
@@ -54,10 +54,13 @@ def requirements(self):
             self.requires("xapian-core/1.4.19")
             self.requires("zlib/1.2.13")
 
-    def package_id(self):
-        del self.info.settings.compiler
-
     def compatibility(self):
+        if (Version(conan_version).major >= 2):
+            # Models libc++ compatibility and identifies Debug builds as equivalent to Release builds
+            compatible_versions = [{"settings": [("compiler.version", v), ("build_type", "Release")]}
+                for v in self.settings.compiler.version.possible_values() if v <= Version(self.settings.compiler.version)]
+            return compatible_versions
```

Comment:
To be clear about the intent of this PR - **this `compatibility()` implementation is not the main purpose of this PR**. The main purpose of this PR is to delete the `del self.info.settings.compiler` statement in the `package_id()` method. The information erasure here hurts the ability for users to manage libcxx abi compatibility.

This `compatibility()` implementation is a _proposal for consideration_ of how to improve the model of libcxx compatibility going forward. It is not perfect (in the sense that it doesn't capture the binary compatibility for all binaries that would be compatible), but it does improve upon the representation indended by deleting `self.info.settings.compiler`.

Unfortunately the effectiveness of this `compatibility()` definition is largely inhibited by restrictions around CCI recipes modifying the `package_id` mode. As it stands, this only provides a benefit if a user were to attempt to consume doxygen as follows, assuming that a binary had been created using the `gcc10` profile already:

```bash
conan install --requires doxygen/1.9.4 -pr:h gcc10 -pr:b gcc10 -s "doxygen/*:compiler.version=gcc12"
```
This is because, if the user does something like this:
```bash
conan install --requires doxygen/1.9.4 -pr:h gcc12 -pr:b gcc12
```
Then the entire dependency tree utilises `gcc12`, and the package id identified for a compatible version of doxygen (built with gcc10) changes.

Given the limited utility of this `compatibility()` implementation, **I'm happy to remove it if required**. It **does** positively contribute to this recipe, but as above, only in a limited way. I wanted to present this for consideration to trigger some thought about when the `compatibility()` method provides value in CCI recipes/what a standard practice should be.